File: problems/283-move-zeroes/solution.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        zero_count = 0
        for i in range(len(nums)):
            if nums[i] == 0:
                zero_count += 1
            else:
                if zero_count > 0:
                    nums[i - zero_count], nums[i] = nums[i], nums[i - zero_count]

File: problems/283-move-zeroes/solution.c
void moveZeroes(int* nums, int numsSize) {
    int nonzero_index = 0;
    for (int i = 0; i < numsSize; i++) {
        if (nums[i] != 0) {
            nums[nonzero_index++] = nums[i];
        }
    }
    for (int i = nonzero_index; i < numsSize; i++) {
        nums[i] = 0;
    }
}